content/logs/languages/php.md

```diff
@@ -0,0 +1,503 @@
+    use Monolog\Handler\StreamHandler;
+    use Monolog\Formatter\JsonFormatter;    
+    ```
+
+
+### Setup - Log into a file with Monolog
+
+The following configuration enables the Json formatting and writes the logs and events into the `application-json.log` file. Edit your code, right after the initialization of the Monolog instance and add a new handler:
+
+```php
+ <?php
+
+require __DIR__ . '/vendor/autoload.php';
+
+// load Monolog library
+use Monolog\Logger;
+use Monolog\Handler\StreamHandler;
+use Monolog\Formatter\JsonFormatter;
+
+// create a log channel
+$log = new Logger('channel_name');
+
+// create a Json formatter
+$formatter = new JsonFormatter();
+
+// create a handler
+$stream = new StreamHandler(__DIR__.'/application-json.log', Logger::DEBUG);
+$stream->setFormatter($formatter);
+
+// bind
+$log->pushHandler($stream);
+
+// an example
+$log->info('Adding a new user', array('username' => 'Seldaek'));
+```
+
+Then [stream your log file to Datadog](/logs/#custom-log-collection)
+
+### Add meta field and context
+
+Much of the usefulness information comes from additional context data that you add to your logs and events.  
+Monolog makes this very convenient by providing methods to set thread local context data that is then submitted automatically with all events. At any moment, log an event with a contextual data:
+
+```php
+ <?php
+
+$logger->info('Adding a new user', array('username' => 'Seldaek'));
+```
+
+But, most important, Monolog comes with a pre-processor feature. It's a simple callback enriches all your events with, for instance, the session id, the request id, and so on:
+
+```php
+ <?php 
+    
+$log->pushProcessor(function ($record) {
+
+    // record the current user
+    $user = Acme::getCurrentUser();
+    $record['context']['user'] = array(
+        'name' => $user->getName(),
+        'username' => $user->getUsername(),
+        'email' => $user->getEmail(),
+    );
+
+    // Add various tags
+    $record['ddtags'] = array('key' => 'value');
+
+    // Add various generic context
+    $record['extra']['key'] = 'value';
+
+    return $record;
+});
+```
+
+### Framework integration 
+
+Monolog is a part of the following framework:
+
+* [Symfony2, Symfony3](/logs/languages/php/#symfony-v2-v3)
+* [PPI](/logs/languages/php/#ppi)
+* [Laravel 4 & 5](/logs/languages/php/#laravel)
+* [Silex](/logs/languages/php/#silex)
+* [Lumen](/logs/languages/php/#lumen)
+* [CakePHP](/logs/languages/php/#cakephp)
+
+Integrate Monolog with your framework then configure your logger: 
+ 
+```php
+ <?php
+
+// Check if the Monolog library is well loaded
+//use Monolog\Logger;
+//use Monolog\Handler\StreamHandler;
+//use Monolog\Formatter\JsonFormatter;
+  
+// with the monolog instance
+$monolog = ...
+  
+///// Log shipper configuration
+
+$formatter = new JsonFormatter();
+$stream = new StreamHandler(__DIR__.'/application-json.log', Logger::DEBUG);
+$stream->setFormatter($formatter);
+
+$monolog->pushHandler($stream);
+return $r;});
```

Comment:
Remove the `});` at the end of the `return` line

> `return $r;`

Reply:
👌